### PR TITLE
Improve testing; run loading of sample

### PR DESCRIPTION
**Pull request by adamdickmeiss**
RestVerticleTest loads sample records again.

### CODE REVIEW

File: src/test/java/org/folio/logintest/CredentialExistenceTest.java [hunk 1]
```diff
@@ -61,26 +56,25 @@ public static void setup(final TestContext context) {
       context.fail(e);
     }
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/CredentialExistenceTest.java:57]: This try-catch is not needed, VertxUnitRunner automatically fails the test if an Exception is thrown.

File: src/test/java/org/folio/logintest/CredentialExistenceTest.java [hunk 2]
```diff
@@ -61,26 +56,25 @@ public static void setup(final TestContext context) {
...
+        new DeploymentOptions().setConfig(new JsonObject().put("http.port", port));
+    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions)
+        .onComplete(context.asyncAssertSuccess(res -> {
+          TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
+          TestUtil.postSync(ta, TENANT, port, vertx)
+              .onComplete(context.asyncAssertSuccess(res2 ->
+                  saveCredential(EXISTING_CREDENTIALS_USER_ID, "password")
+                      .onComplete(context.asyncAssertSuccess())
+              ));
+        }));
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/CredentialExistenceTest.java:77]:
```suggestion
    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions)
        .compose(x -> TestUtil.postSync(ta, TENANT, port, vertx))
        .compose(x -> saveCredential(EXISTING_CREDENTIALS_USER_ID, "password"))
        .onComplete(context.asyncAssertSuccess());
```

File: src/test/java/org/folio/logintest/TestUtil.java [hunk 3]
```diff
@@ -110,4 +115,14 @@ private static void handleResponse(AsyncResult<HttpResponse<Buffer>> ar, HttpMet
       }
     }
   }
+
+  public static Future<Void> postSync(TenantAttributes ta, String tenant, int port, Vertx vertx) {
+    Map<String, String> okapiHeaders = new CaseInsensitiveMap();
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/TestUtil.java:120]:
```suggestion
    Map<String, String> okapiHeaders = new CaseInsensitiveMap<>();
```

File: src/test/java/org/folio/logintest/UpdateCredentialsHistoryTest.java [hunk 4]
```diff
@@ -133,21 +127,17 @@ public void testInitialPasswordIsSavedToHistory(TestContext context) {
 
     Promise<Results<CredentialsHistory>> promise = Promise.promise();
     pgClient.get(TABLE_NAME_CRED_HIST, CredentialsHistory.class, new Criterion(userIdCrit), false, promise);
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/UpdateCredentialsHistoryTest.java:129]: Using the pgClient.get variant that returns a Future avoids the promise.

File: src/test/java/org/folio/logintest/CredentialExistenceTest.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/logintest/CredentialExistenceTest.java]: Yes.

File: src/test/java/org/folio/logintest/TestUtil.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/logintest/TestUtil.java]: ok

File: src/test/java/org/folio/logintest/LogEventsApiTest.java [hunk 5]
```diff
@@ -89,26 +84,17 @@ public static void setUpClass(final TestContext context) {
...
+        .onComplete(context.asyncAssertSuccess(res -> {
       TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
-      TenantAPI tenantAPI = new TenantRefAPI();
-      Map<String, String> okapiHeaders = Map.of("x-okapi-url", "http://localhost:" + port,
-          "x-okapi-tenant", TENANT_ID);
-      tenantAPI.postTenantSync(ta, okapiHeaders, handler -> async.complete(),
-          vertx.getOrCreateContext());
-    });
+      TestUtil.postSync(ta, TENANT_ID, port, vertx).onComplete(context.asyncAssertSuccess());
+    }));
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/LogEventsApiTest.java:91]:
```
    DeploymentOptions restDeploymentOptions = new DeploymentOptions()
      .setConfig(new JsonObject().put(HTTP_PORT, port));
    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
    vertx.deployVerticle(RestVerticle.class.getName(), restDeploymentOptions)
        .compose(x -> TestUtil.postSync(ta, TENANT_ID, port, vertx))
        .onComplete(context.asyncAssertSuccess());
```

File: src/test/java/org/folio/logintest/LoginAttemptsTest.java [hunk 6]
```diff
@@ -99,30 +96,23 @@ public static void setup(final TestContext context) throws Exception {
...
+        .build();
+
+    vertx.deployVerticle(UserMock.class.getName(), mockOptions)
+        .onComplete(context.asyncAssertSuccess(res ->
+            vertx.deployVerticle(RestVerticle.class.getName(), options)
+                .onComplete(context.asyncAssertSuccess(res2 -> {
+                  TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
+                  TestUtil.postSync(ta, TENANT_DIKU, port, vertx).onComplete(context.asyncAssertSuccess());
+                }))
+        ));
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/LoginAttemptsTest.java:115]:
```
    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
    vertx.deployVerticle(UserMock.class.getName(), mockOptions)
        .compose(x -> vertx.deployVerticle(RestVerticle.class.getName(), options))
        .compose(x -> TestUtil.postSync(ta, TENANT_DIKU, port, vertx)
        .onComplete(context.asyncAssertSuccess());
```

File: src/test/java/org/folio/logintest/PasswordRepeatabilityValidationTest.java [hunk 7]
```diff
@@ -74,29 +69,23 @@ public static void setup(final TestContext context) {
...
-      .setContentType(ContentType.JSON)
-      .setBaseUri("http://localhost:" + port)
-      .addHeader(XOkapiHeaders.TENANT, TENANT)
-      .addHeader(XOkapiHeaders.TOKEN, "dummytoken")
-      .addHeader(XOkapiHeaders.URL, "http://localhost:" + port)
-      .build();
+      TestUtil.postSync(ta, TENANT, port, vertx).onComplete(context.asyncAssertSuccess(res2 ->
+        fillInCredentialsHistory().compose(x -> saveCredential()).onComplete(context.asyncAssertSuccess())
+      ));
+    }));
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/PasswordRepeatabilityValidationTest.java:88]:
```
    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions)
        .compose(x -> TestUtil.postSync(ta, TENANT, port, vertx))
        .compose(x -> fillInCredentialsHistory())
        .compose(x -> saveCredential())
        .onComplete(context.asyncAssertSuccess());
```

File: src/test/java/org/folio/logintest/RestVerticleTest.java [hunk 8]
```diff
@@ -381,24 +367,20 @@ public void testAuthnUpdateNoOkapiUrl(TestContext context) {
...
+            doRequest(vertx, credentialsUrl + "?userId=" + userId, HttpMethod.DELETE, null, null,
+                500, "Postgres Error on Delete credentials by userId")
+                .compose(r -> {
+                  TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.0.0");
+                  List<Parameter> parameters = new LinkedList<>();
+                  parameters.add(new Parameter().withKey("loadSample").withValue("true"));
+                  ta.setParameters(parameters);
+                  return TestUtil.postSync(ta, "diku", port, vertx);
+                })
+        );
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/RestVerticleTest.java:383]:
```
    return PostgresClient.getInstance(vertx, "diku")
        .execute("DROP TABLE " + PasswordStorageServiceImpl.TABLE_NAME_CREDENTIALS + " CASCADE")
        .compose(x ->
            doRequest(vertx, credentialsUrl + "?userId=" + userId, HttpMethod.DELETE, null, null,
                500, "Postgres Error on Delete credentials by userId"))
        .compose(x -> {
          TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.0.0");
          List<Parameter> parameters = new LinkedList<>();
          parameters.add(new Parameter().withKey("loadSample").withValue("true"));
          ta.setParameters(parameters);
          return TestUtil.postSync(ta, "diku", port, vertx);
        });
```

File: src/test/java/org/folio/logintest/LogEventsApiTest.java (hunk 5 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/logintest/LogEventsApiTest.java:91]: Oh..Missed more of these.. Will do them all .

File: src/test/java/org/folio/logintest/LogEventsApiTest.java [hunk 9]
```diff
@@ -80,35 +75,21 @@ public static void setUpClass(final TestContext context) {
...
-      TenantAPI tenantAPI = new TenantRefAPI();
-      Map<String, String> okapiHeaders = Map.of("x-okapi-url", "http://localhost:" + port,
-          "x-okapi-tenant", TENANT_ID);
-      tenantAPI.postTenantSync(ta, okapiHeaders, handler -> async.complete(),
-          vertx.getOrCreateContext());
-    });
+    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
+    vertx.deployVerticle(RestVerticle.class.getName(), restDeploymentOptions)
+        .compose(res -> TestUtil.postSync(ta, TENANT_ID, port, vertx))
+            .onComplete(context.asyncAssertSuccess());
```
Comment (julianladisch) [inline src/test/java/org/folio/logintest/LogEventsApiTest.java:86]:
```suggestion
        .onComplete(context.asyncAssertSuccess());
```

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

File: src/test/java/org/folio/logintest/LogEventsApiTest.java (hunk 9 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/logintest/LogEventsApiTest.java]: Stop Julian!

**Pull request merged**